### PR TITLE
Support non-pointer JSONWebKey in NewSigner

### PR DESCRIPTION
**Pull request by shaxbee**

### CODE REVIEW

**Comment by ericchiang** [comment]
Verify as well?

**Comment by shaxbee** [comment]
Added Verify / NewEncrypter / Decrypt

File: signing.go [hunk 1]
```diff
@@ -105,6 +105,8 @@ func newVerifier(verificationKey interface{}) (payloadVerifier, error) {
 		return &symmetricMac{
 			key: verificationKey,
 		}, nil
+	case JSONWebKey:
+		return newVerifier(verificationKey.key)
```
Comment (csstaub) [inline signing.go]: Should be `verificationKey.Key` here

**Comment by csstaub** [comment]
LGTM, but looks like you have a small typo which is causing the build to fail.

**Comment by csstaub** [comment]
Looks good -- thank you @shaxbee.

**Pull request merged**